### PR TITLE
InnerProtocolParticipant message other participants

### LINKED ISSUE #286: Add `InnerProtocolParticipant` method to create a message for all participants with the same content.
We often see this pattern in the `...Participant` code:
```rust
self.other_participant_ids
            .iter()
            .map(|&other_participant_id| {
                Message::new(
                    MessageType::Keygen(KeygenMessageType::R2Decommit),
                    sid,
                    self.id,
                    other_participant_id,
                    &decom_bytes,
                )
            })
            .collect();
```
It would be better if this was incapsulated in a method within `InnerProtocolParticipant`, such as:
```rust
pub(crate) fn message_for_other_participants<T: Serializable>(message_type: MessageType, data: T)
```

## Closing criteria
- [x] Implement method for the `InnerProtocolParticipant` trait
- [x] Replace all the instances of this pattern across `{Keygen | Auxinfo | Presign}Participant`
- [ ] Add a test that checks that the messages are identical except for the `to` field for the whole returned list?

### PR DESCRIPTION
**Pull request by hridambasu**
Closes #286

### CODE REVIEW

**Comment by gatoWololo** [comment]
@hridambasu I found the bug plus a couple of comments :)
- The branch is currently called `ipp-msg-oth-part`. Usually we use the convention of having the issue number first. So something like `286-ipp-msg-oth-part`. This makes it easier to find the branch.
- Your `message_for_other_participants` function takes a `from: ParticipantIdentifier,` but that is always passed as `self.id`. I suggest just using `self.id()` in this method and getting rid of the `from` argument.

File: src/broadcast/participant.rs [hunk 1]
```diff
@@ -325,22 +318,16 @@ impl BroadcastParticipant {
...
-                    &data_bytes,
-                )
-            })
-            .collect();
+        let messages = self.message_for_other_participants(
+            MessageType::Broadcast(BroadcastMessageType::Redisperse),
+            message.id(),
+            self.id,
+            data,
+        )?;
```
Comment (gatoWololo) [inline src/broadcast/participant.rs:330]: The bug is here. Notice before we were iterating over all entries in `others_minus_leader` which is just `other_participant_ids` minus the `leader` ID. Your current version iterates over everything _including_ the leader.

I would leave your current code as is, that generates messages for all participants, but after generating the messages, call the `retain` function that filters out the leader. Just how the old code does. Lmk if you have any questions about this.

**Review by gatoWololo — APPROVED** [review]
Looks good to me!

**Comment by gatoWololo** [comment]
Definitely squash the commits before merging!

**Comment by marsella** [comment]
It looks like in the place where you were getting a bug (making `Redisperse` messages for every party in broadcast), you ended up not using the new function. I think @gatoWololo was suggesting that you _do_ use the new function, but take out the "leader" from the message set (rather than from the participant set like was happening previously).

**Comment by hridambasu** [comment]
Yeah. But I checked the original issue and saw that @amaloz only asked to replace the particular pattern in the code in the 3 modules of AuxInfo, Presign and Keygen. So I resisted tampering with any other part of the code and put back the code in its previous form in broadcast. If you want me to use it as @gatoWololo suggested, then I can write a separate PR for changing that. :)

**Comment by marsella** [comment]
I see. That's a little inconsistent because it is used for the broadcast `Disperse` message, just not the ``Redisperse`. But it's fine; I think it'll get superseded in broadcast when we get to #374 anyway.

**Pull request merged**